openpype/hosts/maya/api/lib.py

```diff
@@ -3446,3 +3446,17 @@ def dgcontext(mtime):
 def get_attribute_input(attr):
     connections = cmds.listConnections(attr, plugs=True, destination=False)
     return connections[0] if connections else None
+
+
+def write_xgen_file(data, filepath):
+    lines = []
+    with open(filepath, "r") as f:
+        for line in [line.rstrip() for line in f]:
+            for key, value in data.items():
+                if line.startswith("\t{}".format(key)):
```

Comment:
Good point! Trailing with `\t\t` before the value so adding that.

Reply:
By the way, if that's the case - we could rewrite this to something like this:

```python
def write_xgen_file(data, filepath):

    # Generate regex lookup for line to key basically
    # match any of the keys in `\t{key}\t\t`
    keys = "|".join(re.escape(key) for key in data.keys())
    re_keys = re.compile("^\t({})\t\t".format(keys))
    
    lines = []
    with open(filepath, "r") as f:
        for line in f:
            match = re_keys.match(line)
            if match:
                key = match.group(1)
                value = data[key]
                line = "\t{}\t\t{}".format(key, value)

            lines.append(line)

    with open(filepath, "w") as f:
        f.write("\n".join(lines))
```

This would be much faster if the file being iterated tends to exist of many lines and the amount of keys in data isn't super small.

Reply:
Cool, thanks. Will ingest and test